dockstore-webservice/src/test/java/io/dockstore/webservice/permissions/sam/SamPermissionsImplTest.java

```diff
@@ -490,6 +492,35 @@ public void testSelfDestruct() throws ApiException {
 
     }
 
+    @Test
+    public void testSetPermissionWhenResourceCreatedWithoutAllPolicies() throws ApiException {
+        when(fooWorkflow.getUsers()).thenReturn(new HashSet<>(Arrays.asList(userMock)));
```

Comment:
these probably need a few more comments to explain whats going on
this looks like it tests that all (missing) default policies re-created if permissions are set

Reply:
Agreed on need for comment.

When you create a resource in SAM, it only adds the owner policy to it.

So when we invoke SAM API to create resource, we [immediately add the writer and reader policies](https://github.com/ga4gh/dockstore/blob/feature/1713/initpermissions/dockstore-webservice/src/main/java/io/dockstore/webservice/permissions/sam/SamPermissionsImpl.java#L291) to it.

This is testing the case where somebody went into the SAM API and only created the resource, but didn't add those policies, which our code relies on existing when setting a permission.